lib/kitchen/driver/openstack.rb

```diff
@@ -150,6 +167,13 @@ def create_server
         # Can't use the Fog bootstrap and/or setup methods here; they require a
         # public IP address that can't be guaranteed to exist across all
         # OpenStack deployments (e.g. TryStack ARM only has private IPs).
+        check_server(server_def)
+      end
+
+      def check_server(server_def)
+        compute.servers.each do |server|
+          server if server.name == server_def[:name]
```

Comment:
The description mentions ignoring servers with names that already exist but, if I'm reading this right, it's not doing that. `compute.servers.create` is still being called after this loop regardless of what it returns.

Also, is skipping a server with a name collision really a desirable behavior? Seems like that could cause issues if, for example, two people on the same development team were trying to run Test Kitchen at the same time (and their project used configured names instead of randomly generated ones).